### PR TITLE
fix: get credentials from credential_process in aws profile config

### PR DESCRIPTION
**Pull request by edwardfoyle**
#### Description of changes

We had some logic to handle AWS credential_process profiles, but we were not actually calling the credential process (or at least not in all cases). This change calls the credential process when we are initially resolving credentials.

I updated some error handling to use AmplifyError and remove some calls to process.exit because it caused some hanging spinners if the credential process fails.

There is also a change to respect the `AWS_CONFIG_FILE` and `AWS_SHARED_CREDENTIALS_FILE` environment variables when resolving the path to the aws config and credentials files.

There are also an unfortunate amount of lint fixes :(

#### Issue #, if available
fixes https://github.com/aws-amplify/amplify-cli/issues/11319 https://github.com/aws-amplify/amplify-cli/issues/4488

#### Description of how you validated changes
Manually tested, updated existing unit tests and running e2e now: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli?branch=run-e2e/foyleef/cred-process

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/dev/CONTRIBUTING.md#tests)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli-core/src/state-manager/pathManager.ts [hunk 1]
```diff
@@ -194,31 +178,33 @@ export class PathManager {
...
+  getCustomPoliciesPath = (category: string, resourceName: string): string =>
+    path.join(this.getResourceDirectoryPath(undefined, category, resourceName), PathConstants.CustomPoliciesFilename);
 
-  getAWSCredentialsFilePath = (): string => path.normalize(path.join(this.getDotAWSDirPath(), PathConstants.AWSCredentials));
+  getAWSCredentialsFilePath = (): string =>
+    process.env.AWS_SHARED_CREDENTIALS_FILE || path.normalize(path.join(this.getDotAWSDirPath(), PathConstants.AWSCredentials));
 
-  getAWSConfigFilePath = (): string => path.normalize(path.join(this.getDotAWSDirPath(), PathConstants.AWSConfig));
+  getAWSConfigFilePath = (): string =>
+    process.env.AWS_CONFIG_FILE || path.normalize(path.join(this.getDotAWSDirPath(), PathConstants.AWSConfig));
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/src/state-manager/pathManager.ts:207]: This is the only functional change in this file

File: packages/amplify-cli-core/src/state-manager/pathManager.ts [hunk 2]
```diff
@@ -95,31 +95,25 @@ export class PathManager {
 
   getAmplifyAdminDirPath = (): string => this.constructPath(this.getHomeDotAmplifyDirPath(), [PathConstants.AmplifyAdminDirName]);
 
-  getAmplifyAdminConfigFilePath = (): string => this.constructPath(
-    this.getAmplifyAdminDirPath(), [PathConstants.AmplifyAdminConfigFileName],
-  );
+  getAmplifyAdminConfigFilePath = (): string =>
```
Comment (edwardfoyle) [inline packages/amplify-cli-core/src/state-manager/pathManager.ts:98]: this file is all linting changes except for the lines noted below

File: packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts [hunk 3]
```diff
@@ -22,31 +25,57 @@ describe('profile tests', () => {
...
-    expect(profile_config.credentialProvider).toBeDefined();
-    expect(profile_config.credentialProvider).toBeInstanceOf(aws.CredentialProviderChain);
-    expect(getProfileCredentials_mock).toHaveBeenCalledTimes(0);
+    expect(profile_config.accessKeyId).toBe('testAccessKey');
+    expect(profile_config.secretAccessKey).toBe('testSecret');
+    expect(profile_config.sessionToken).toBe('testSessionToken');
+    expect(profile_config.expiration).toEqual(new Date(1234));
   });
 
   it('should fail to return profiled aws credentials', async () => {
```
Comment (edwardfoyle) [inline packages/amplify-provider-awscloudformation/src/__tests__/system-config-manager.test.ts:66]: the changes to this test are just linting

**Review by jhockett — APPROVED** [review]
LGTM, pending passing e2e run

**Pull request merged**